crates/core/src/db/datastore/locking_tx_datastore/mut_tx.rs

```diff
@@ -1196,35 +1196,27 @@ impl MutTxId {
     pub(super) fn insert_row_internal(&mut self, table_id: TableId, row: &ProductValue) -> Result<RowRefInsertion<'_>> {
         let commit_table = self.committed_state_write_lock.get_table(table_id);
 
-        // Check for constraint violations as early as possible,
-        // to ensure that `UniqueConstraintViolation` errors have precedence over other errors.
-        // `tx_table.insert` will later perform the same check on the tx table,
-        // so this method needs only to check the committed state.
-        if let Some(commit_table) = commit_table {
-            commit_table
-                .check_unique_constraints(row, |maybe_conflict| self.tx_state.is_deleted(table_id, maybe_conflict))
-                .map_err(IndexError::from)?;
-        }
-
         // Get the insert table, so we can write the row into it.
         let (tx_table, tx_blob_store, _, delete_table) = self
             .tx_state
             .get_table_and_blob_store_or_maybe_create_from(table_id, commit_table)
             .ok_or(TableError::IdNotFoundState(table_id))?;
 
         match tx_table.insert(tx_blob_store, row) {
-            Ok((hash, row_ref)) => {
-                // `row` not previously present in insert tables,
-                // but may still be a set-semantic conflict with a row
-                // in the committed state.
-
-                let ptr = row_ref.pointer();
+            Ok((tx_row_hash, tx_row_ref)) => {
+                let tx_row_ptr = tx_row_ref.pointer();
                 if let Some(commit_table) = commit_table {
-                    // Safety:
+                    // The `tx_row_ref` was not previously present in insert tables,
+                    // but may still be a set-semantic conflict with a row
```

Comment:
Comment should be expanded to say that it may also be a constraint violation. I'm on mobile RN and can't find the suggestion button, but would add "or may violate a constraint" after this sentence.